Atomic/syscontainers.py

```diff
@@ -407,10 +408,28 @@ def _generate_default_oci_configuration(self, destination):
         with open(conf_path, 'w') as conf:
             conf.write(json.dumps(configuration, indent=4))
 
-    def _generate_systemd_startstop_directives(self, name, pidfile=None, can_detach=False):
+    def _generate_systemd_startstop_directives(self, name, pidfile=None, unit_file_support_pidfile=False):
         if self.user:
-            return ["%s '%s'" % (util.BWRAP_OCI_PATH, name), "", "", ""]
+            return self._generate_user_systemd_startstop_directives(name, pidfile, unit_file_support_pidfile)
+        else:
+            return self._generate_system_systemd_startstop_directives(name, pidfile, unit_file_support_pidfile)
 
+    # --user case
+    def _generate_user_systemd_startstop_directives(self, name, pidfile=None, unit_file_support_pidfile=False):
+        if unit_file_support_pidfile:
+            has_pidfile_option = False
+            try:
+                has_pidfile_option = "--pid-file" in str(util.check_output([util.BWRAP_OCI_PATH, "--help"], stderr=DEVNULL))
```

Comment:
yes we could do that.  Just there is no bwrap/bwrap-oci release yet so it would break with existing versions.  I thought of changing it once we have a new version released.